aw-server/src/endpoints/settings.rs

```diff
@@ -0,0 +1,88 @@
+use crate::endpoints::ServerState;
+use rocket::http::Status;
+use rocket::State;
+use rocket_contrib::json::{Json, JsonValue};
+use std::sync::MutexGuard;
+
+use aw_datastore::{Datastore, DatastoreError};
+
+fn parse_key(key: String) -> Result<String, Status> {
+    let namespace: String = "settings.".to_string();
+    if key.len() >= 128 {
+        return Err(Status::BadRequest);
+    } else {
+        Ok(namespace + key.as_str())
+    }
+}
+
+#[post("/<key>", data = "<message>")]
+pub fn setting_new(
+    state: State<ServerState>,
+    key: String,
+    message: Json<String>,
+) -> Result<Status, Status> {
+    let setting_key = match parse_key(key) {
```

Comment:
You don't have to match it like that, you can just do the following on a single like.

    let setting_key = parse_key(key)?;

The ? operator is nice for removing this kind of boilerplate code.

Reply:
Good point. I already used it at some of the newer changes but probably missed some.